quinn-proto/src/config.rs

```diff
@@ -404,6 +404,8 @@ where
 
     /// Private key used to authenticate data included in handshake tokens.
     pub(crate) token_key: Arc<S::HmacKey>,
+
+    pub(crate) master_key: Arc<Vec<u8>>,
```

Comment:
We don't need two separate keys; after these changes `token_key` is unused. Rather than replacing `token_key`, I suggest using it as the master key directly.